pep-0722.rst

```diff
@@ -17,87 +17,136 @@ Resolution:
+
 
-This inability to determine the negative case makes ``TypeGuard`` not as useful as
-it could be.
+This PEP proposes that when the type argument of the ``TypeGuard`` is a subtype
+of the type of the first input argument, then the false case can be further
+narrowed. 
 
-This PEP proposes that in cases where the output type is a *strict* subtype of
-the input type, the negative case can be computed. This changes the example so
-that the ``int`` case is possible:
-["output type" -- might need to define this term or use something else. I don't see that term used in PEP 647.]
-["This changes the example" -- maybe rephrase this to clarify that the code of the example is unchanged, but type checkers can interpret it differently?]
-["is possible" seems pretty vague]
-[What does strict subtype mean? And why is it italicized?]
+This changes the example above like so:
 
-::
+.. code-block:: python
 
     def is_str(val: str | int) -> TypeGuard[str]:
         return isinstance(val, str)
 
     def func(val: str | int):
         if is_str(val):
-            reveal_type(val) # str
+            # Type checkers can assume val is a 'str' in this branch
         else:
-            reveal_type(val) # int
+            # Type checkers can assume val is an 'int' in this branch
 
-Since the output type is a *strict* subtype of the
-input, a type checker can determine that the only possible type in the ``else`` is the
-other input type(s).
-["the other input type(s)" -- There's only one input type. It's a Union. Suggest rephrasing this. I'm not sure if talking about the types using set theory (input -- output) would make this more clear (or more generic) or worse.]
+Since the ``TypeGuard`` type (or output type) is a subtype of the input argument
+type, a type checker can determine that the only possible type in the ``else``
+is the other type in the Union. In this example, it is safe to assume that if
+``is_str`` returns false, then type of the ``val`` argument is an ``int``.
 
-If the output type is not a *strict* subtype of the input type,
-the negative cannot be assumed to be the intuitive opposite:
-["intuitive opposite" -- opposite is the incorrect term here and I think intuition doesn't belong in a PEP :)]
+Unsafe Narrowing
+--------------------
 
-::
+There are cases where this further type narrowing is not possible. Here's an
+example:
+
+.. code-block:: python
 
     def is_str_list(val: list[int | str]) -> TypeGuard[list[str]]
         return all(isinstance(x, str) for x in val)
 
     def func(val: list[int | str]):
         if is_str_list(val):
-            reveal_type(val) # list[str]
+            # Type checker assumes list[str] here
+        else:
+            # Type checker cannot assume list[int] here
+
+Since ``list`` is invariant, it doesn't have any subtypes. This means type
+checkers cannot narrow the type to ``list[int]`` in the false case.
+``list[str]`` is not a subtype of ``list[str | int]``.
```

Comment:
Suggest moving the last sentence after the first sentence.

```suggestion
Since ``list`` is invariant, it doesn't have any subtypes. ``list[str]`` is not a subtype of ``list[str | int]``. This means type
checkers cannot narrow the type to ``list[int]`` in the false case.
```